pkg/service/one2onerestore/progress.go

```diff
@@ -0,0 +1,339 @@
+
+		ClusterID: w.runInfo.ClusterID,
+		TaskID:    w.runInfo.TaskID,
+		RunID:     w.runInfo.RunID,
+
+		KeyspaceName:     table.Keyspace,
+		TableName:        table.Table,
+		RemoteSSTableDir: remoteDir,
+		AgentJobID:       jobID,
+		Host:             host,
+		ShardCnt:         shardCount,
+		TableSize:        table.Size,
+		StartedAt:        &started,
+	}
+
+	if err := w.insertRunProgress(ctx, &pr); err != nil {
+		w.logger.Error(ctx, "Download progress", "err", err, "pr", pr)
+	}
+	return &pr
+}
+
+func (w *worker) updateDownloadProgress(ctx context.Context, pr *RunProgress, job *scyllaclient.RcloneJobProgress) {
+	startedAt, completedAt := time.Time(job.StartedAt), time.Time(job.CompletedAt)
+	if !startedAt.IsZero() {
+		pr.StartedAt = &startedAt
+	}
+	if !completedAt.IsZero() {
+		pr.CompletedAt = &completedAt
+	}
+	pr.Error = job.Error
+	pr.Downloaded = job.Uploaded
+	pr.Skipped = job.Skipped
+	pr.Failed = job.Failed
+
+	if err := w.insertRunProgress(ctx, pr); err != nil {
+		w.logger.Error(ctx, "Update download progress", "err", err, "pr", pr)
+	}
+}
+
+func (w *worker) progressDone(ctx context.Context, start, end time.Time) {
+	pr := RunProgress{
+		Stage: StageDone,
+
+		ClusterID: w.runInfo.ClusterID,
+		TaskID:    w.runInfo.TaskID,
+		RunID:     w.runInfo.RunID,
+
+		StartedAt:   &start,
+		CompletedAt: &end,
+	}
+	if err := w.insertRunProgress(ctx, &pr); err != nil {
+		w.logger.Error(ctx, "Done progress", "err", err, "pr", pr)
+	}
+}
+
+func (w *worker) insertRunProgress(ctx context.Context, pr *RunProgress) error {
+	// The main reason for these checks is the ability to 'mock' this function simply by
+	// passing empty RunProgress.
+	if pr.ClusterID == uuid.Nil || pr.TaskID == uuid.Nil || pr.RunID == uuid.Nil {
+		return errors.New("ClusterID, TaskID and RunID can't be empty")
+	}
+	q := table.One2onerestoreRunProgress.InsertQueryContext(ctx, w.managerSession).BindStruct(pr)
+	return q.ExecRelease()
+}
+
+func (w *worker) getProgress(ctx context.Context, target Target) (Progress, error) {
+	q := table.One2onerestoreRunProgress.SelectQueryContext(ctx, w.managerSession)
+	iter := q.BindMap(qb.M{
+		"cluster_id": w.runInfo.ClusterID,
+		"task_id":    w.runInfo.TaskID,
+		"run_id":     w.runInfo.RunID,
+	}).Iter()
+
+	pr := w.aggregateProgress(iter, target.SnapshotTag)
+
+	if err := iter.Close(); err != nil {
+		return Progress{}, errors.Wrap(err, "close iterator")
+	}
+	return pr, nil
+}
+
+type dbIterator interface {
+	StructScan(v any) bool
+}
+
+func (w *worker) aggregateProgress(iter dbIterator, snapshotTag string) Progress {
+	var (
+		tablesProgress = map[scyllaTable]TableProgress{}
+		hostsProgress  = map[string]HostProgress{}
+		views          []View
+		pr             RunProgress
+		latestRP       RunProgress
+		resultProgress progress
+	)
+
+	for iter.StructScan(&pr) {
+		latestRP = latestStage(latestRP, pr)
+		resultProgress = incrementProgress(resultProgress, toProgress(pr))
+		switch pr.Stage {
+		case StageData:
```

Comment:
Could you please add `default` case to just log progress entry with non supported stage name ? 

The full list consist of:
```
const (
	StageDropViews     Stage = "DROP_VIEWS"
	StageAlterTGC      Stage = "ALTER_TGC"
	StageData          Stage = "DATA"
	StageRecreateViews Stage = "RECREATE_VIEWS"
	StageDone          Stage = "DONE"
)
```